dlpar/dlpar_workload_setup.py

```diff
@@ -24,35 +24,184 @@
+            else:
+                return False
         except Exception:
             return False
 
+    # HTX Test start
+    def start_htx_engine(self):
+        """
+        Execute 'HTX' with appropriate parameters.
+        """
+        flag = ""
+        self.log.info("selecting the mdt file")
+        cmd = "htxcmdline -select -mdt %s" % mdt_file
+        self.htx_select_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.htx_select_cmd, ignore_status=True)
+        time.sleep(10)
+        self.log.info("Activating the %s", mdt_file)
+        cmd = "htxcmdline -activate -mdt %s" % mdt_file
+        self.htx_act_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.htx_act_cmd, ignore_status=True)
+        time.sleep(10)
+        self.log.info("Running the HTX ")
+        cmd = "htxcmdline -run  -mdt %s" % mdt_file
+        self.htx_runmdt_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.htx_runmdt_cmd, ignore_status=True)
+        time.sleep(5)
+
+    def htxerr_check_engine(self):
+        """
+        Checks if HTX is running, and if no errors.
+        """
+        flag = ""
+        for _ in range(0, self.time_limit, 60):
+            self.log.info("HTX Error logs")
+            cmd = 'htxcmdline -geterrlog'
+            self.htx_geterr_cmd = self.cmd_builder(
+                self.linux_1_pw, self.primary_lpar, cmd, flag)
+            process.system(self.htx_geterr_cmd, ignore_status=True)
+            time.sleep(5)
+            cmd = 'htxcmdline -query  -mdt %s' % mdt_file
+            self.htx_query_cmd = self.cmd_builder(
+                self.linux_1_pw, self.primary_lpar, cmd, flag)
+            process.system(self.htx_query_cmd, ignore_status=True)
+            time.sleep(60)
+
+    def stop_htx_engine(self):
+        '''
+        Shutdown the mdt file and the htx daemon and set SMT to original value
+        '''
+        self.stop_htx()
+
+    def stop_htx(self):
+        """
+        Stop the HTX Run
+        """
+        flag = "HTX"
+        self.log.info("shutting down the %s ", mdt_file)
+        cmd = 'htxcmdline -shutdown -mdt %s' % mdt_file
+        self.shut_downmdt_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.shut_downmdt_cmd, timeout=120, ignore_status=True)
+        cmd = '/usr/lpp/htx/etc/scripts/htxd_shutdown'
+        self.shut_down_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.shut_down_cmd, ignore_status=True)
+        time.sleep(5)
+        cmd = 'umount /htx_pmem*'
+        self.umount_htx_cmd = self.cmd_builder(
+            self.linux_1_pw, self.primary_lpar, cmd, flag)
+        process.system(self.umount_htx_cmd, shell=True, ignore_status=True)
+
+    def get_dist_name_version(self):
```

Comment:
Use the existing avocado library function to get distro name and version

Reply:
@sacsant , Below are the steps for the DLPAR test case,
The host machine runs test scripts
	-> This is where we provide the details of the target LPARS where the actual DLPARS operation commands are being executed (Primary and Secondary LPARS).
	-> For more information, please refer to the following diagram
	-> We are using the SSH protocol to run the DLPAR add, remove, and move commands to target machines (Primary and Secondary LPAs) from our host computer.
![DLPAR](https://user-images.githubusercontent.com/99895065/203007873-2df74397-d4ed-47cc-be27-4e36774a6929.jpg)

-> This is an attempt to get the distro name and version from a remote machine rather than a local one.
I-> nstead of using the avocado library functions to get the target lpar distro name and version, I added the above logic.
**Note:** Avocado library function can be used to retrieve the distro name and version if the avocado test script is running on the same test machine.

Reply:
Below are the steps for the DLPAR test case,
The host machine runs test scripts
	-> This is where we provide the details of the target LPARS where the actual DLPARS operation commands are being executed (Primary and Secondary LPARS).
	-> For more information, please refer to the following diagram
	-> We are using the SSH protocol to run the DLPAR add, remove, and move commands to target machines (Primary and Secondary LPAs) from our host computer.
-> This is an attempt to get the distro name and version from a remote machine rather than a local one.
![DLPAR](https://user-images.githubusercontent.com/99895065/203011993-f5ce079d-f2ec-4d23-9435-22865bb82f26.jpg)
-> Instead of using the avocado library functions to get the target lpar distro name and version, I added the above logic.
**Note**: Avocado library function can be used to retrieve the distro name and version if the avocado test script is running on the same test machine.

Would you please guide me if there is a way to use the avocado library function to obtain the target lpar distro's name and version?

Reply:
> -> This is an attempt to get the distro name and version from a remote machine rather than a local one.
Is this required to setup HTX? If yes then as mentioned earlier if the HTX setup can be done outside of this test then we don't need this function ??

Reply:
Ok, I will try to check how we can use the config file to setup the HTX workload on the remote.

Reply:
> Ok, I will try to check how we can use the config file to setup the HTX workload on the remote.

@abdhaleegit @sacsant as this is happing in remotely so calling cfg will not direct mapping, In this case we need to evolve avocado library case in this case may some of part of code can be put in avocado but i feel that can be pursue as enhancment

Reply:
> Use the existing avocado library function to get distro name and version

@sacsant we don't  have the support of this but seems a good idea maybe some of the network tests also need that where we run commands remotely, @SamirMulani  we should be plan this